includes/lco-functions.php

```diff
@@ -190,8 +190,6 @@ function wc_ledyer_confirm_ledyer_order( $order_id, $ledyer_pending ) {
 
 		do_action( 'ledyer_process_payment', $order_id, $request );
 
-		$ledyer_order = ledyer()->api->update_order_reference( $payment_id, array( 'reference' => strval( $order_id ) ) );
-
```

Comment:
Moved setting order reference to inside where the receive the notification from Ledyer BE, instead of when the confirmation page is triggered by FE.